tests/test_tztrout.py

```diff
@@ -0,0 +1,433 @@
+import datetime
+
+import pytest
+import tztrout
+from mock import patch
+
+
+class FakeDateTime(datetime.datetime):
+    "A datetime replacement that lets you set utcnow()"
+
+    @classmethod
+    def utcnow(cls, *args, **kwargs):
+        if hasattr(cls, 'dt'):
+            return cls.dt
+        raise NotImplementedError(
+            'use FakeDateTime.set_utcnow(datetime) first'
+        )
+
+    @classmethod
+    def set_utcnow(cls, dt):
+        cls.dt = dt
+
+
+class TestTZTrout:
+    def assert_only_one_tz(self, ids, tz_name, tz_names):
+        """ Assert that a given set of timezone ids only matches one tz name
+         in a given set of tz names
+         """
+        tz_names_copy = tz_names[:]
+        assert tz_name in tz_names_copy
+        tz_names_copy.remove(tz_name)
+        assert set(tztrout.tz_ids_for_tz_name(tz_name)) & set(ids)
+        for other_name in tz_names_copy:
+            assert not (set(tztrout.tz_ids_for_tz_name(other_name)) & set(ids))
```

Comment:
This is quite a convoluted assertion (and I know it's been here before, but now seems to be the time to untangle this web). What are we actually using it for? Is there a less complex way to assert the correctness of whatever we're testing?

Reply:
I rewrote this method in 22a8186 to use less complexity and more straightforward variable names and comments.

Reply:
I still dislike that we test things in this way (we should separately test the return values of `tz_ids_for_tz_name`, `tz_ids_for_phone`, and `tz_ids_for_address`), but that can be a future improvement and at least what we have is decently readable now. Well done!